integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/NewCarbonDataLoadRDD.scala

```diff
@@ -316,6 +319,7 @@ class NewDataFrameLoaderRDD[K, V](
             carbonLoadModel.getTableName,
             carbonLoadModel.getSegment.getSegmentNo))
         executor.execute(model, loader.storeLocation, recordReaders.toArray)
+        executor.close()
```

Comment:
good catch.

But better to add it inside taskCompletion listener. refer `UpdateDataLoad.scala` line 70

Reply:
it already added, but we need to invoke it before upload metrics.